pkg/cloud/context.go

```diff
@@ -43,4 +43,6 @@ type CallContextKey struct {
 	Version meta.Version
 	// Service is the service being invoked (e.g. "Firewalls", "BackendServices")
 	Service string
+	// Key is the key of the resource being operated on.
+	Key *meta.Key
```

Comment:
Can you rename it to a ResourceKey? CallContextKey.Key is not descriptive enough

Reply:
+1, probably call it `Resource` to avoid stuttering.

Reply:
So...to begin with `Key` is already a strange name given that it represents a location for a resource, and not the resource itself.  I named it `Key` here for consistency since it's probably better to have a bad but consistent name than to use an arbitrary name because it clashes with names in the current package.  If we have a problem with `Key`, I think we need to alias / rename that type in the meta package to something more descriptive like `Location` or `Scope`.

Reply:
renamed to `Resource`